setup.py

```diff
@@ -589,284 +596,300 @@ def get_blosc_version(headername):
+        (blosc_package, BLOSC_DIR),
+    ]:
+
+        if package.tag == "LZO" and lzo2_enabled:
+            print(
+                f"* Skipping detection of {lzo1_package.name} "
+                f"since {lzo2_package.name} has already been found."
+            )
             continue  # do not use LZO 1 if LZO 2 is available
 
         # if a package location is not specified, try to find it in conda env
         if not location and CONDA_PREFIX:
             location = CONDA_PREFIX
 
-        # looking for lzo/lzo1x.h but pkgconfig already returns '/usr/include/lzo'
-        use_pkgconfig = USE_PKGCONFIG if package.tag != 'LZO2' else False
+        # looking for lzo/lzo1x.h but pkgconfig already returns
+        # '/usr/include/lzo'
+        use_pkgconfig = USE_PKGCONFIG and package.tag != "LZO2"
 
         (hdrdir, libdir, rundir) = package.find_directories(
-            location, use_pkgconfig=use_pkgconfig)
+            location, use_pkgconfig=use_pkgconfig
+        )
 
         # check if HDF5 library uses old DLL naming scheme
-        if hdrdir and package.tag == 'HDF5':
-            hdf5_header = os.path.join(hdrdir, "H5public.h")
+        if hdrdir is not None and hdrdir.is_dir() and package.tag == "HDF5":
+            hdf5_header = hdrdir / "H5public.h"
             hdf5_version = get_hdf5_version(hdf5_header)
             if hdf5_version < min_hdf5_version:
                 exit_with_error(
-                    "Unsupported HDF5 version! HDF5 v%s+ required. "
-                    "Found version v%s" % (min_hdf5_version, hdf5_version))
+                    f"Unsupported HDF5 version! HDF5 v{min_hdf5_version}+ "
+                    f"required. Found version v{hdf5_version}"
+                )
 
-            if os.name == 'nt' and hdf5_version < "1.8.10":
+            if os.name == "nt" and hdf5_version < LooseVersion("1.8.10"):
                 # Change in DLL naming happened in 1.8.10
-                hdf5_old_dll_name = 'hdf5dll' if not debug else 'hdf5ddll'
+                hdf5_old_dll_name = "hdf5dll" if not debug else "hdf5ddll"
                 package.library_name = hdf5_old_dll_name
                 package.runtime_name = hdf5_old_dll_name
-                _platdep['HDF5'] = [hdf5_old_dll_name, hdf5_old_dll_name]
+                _platdep["HDF5"] = [hdf5_old_dll_name, hdf5_old_dll_name]
                 _, libdir, rundir = package.find_directories(
-                    location, use_pkgconfig=USE_PKGCONFIG)
+                    location, use_pkgconfig=USE_PKGCONFIG
+                )
 
         # check if the library is in the standard compiler paths
         if not libdir and package.target_function:
-            libdir = compiler.has_function(package.target_function,
-                                        libraries=(package.library_name,))
+            libdir = compiler.has_function(
+                package.target_function, libraries=(package.library_name,)
+            )
 
         if not (hdrdir and libdir):
```

Comment:
if ``hdrdir`` and ``libdirs`` are ``Path``s this is always ``True``, please check